Week-1/Assignment1.md

```diff
@@ -0,0 +1,131 @@
+                          However, since resources are shared, performance can be affected if other websites on the same server experience 
+                          heavy traffic.
+   
+      2. Vps hosting:- Virtual Private Server is like shared hosting but you will have more resource and flexibility then shared hosting, 
+                       and you get a virtual server dedicated to your website. It's suitable for websites that require more resources or   
+                        customization but don't need an entire dedicated server.
+
+      3.Dedicated hosting:- Here you get a physical server dedicated to your website, offering highest use of performance and    
+                            constomization. It's suitable for websites with high traffic, resource-intensive applications, or specific 
+                            customization requirements.
+
+      4. Cloud hosting:- It uses multiple interconnected servers to provide scalability and reliability. It's suitable for websites with 
+                         fluctuating traffic and resource needs.
+      
+      5. Managed Wordpress hosting:- The hosting provider take care of the updates, backups and security. It's suitable when you run a 
+                                   wordpress site and want to focus on content creation without dealing with technical stuff.
+      
+      6. Reseller hosting:- Here hosting allows you to create your own hosting plans and resell them to clients.It's suitable for 
+                            individuals or businesses looking to offer hosting services without managing the server infrastructure 
+                            themselves.
+
+
+## What is scaling, and why is it important for web applications? How does scaling differ for vertical and horizontal scaling?
+
+Scaling websrevers means process of increasing a website capaity to handle more users, traffic and requests. 
+
+Scalind is important as a website grows in popularity it may experience increased load, leading slower response time which leads to crashes. Scaling helps to ensure that website remains accessiable and responsive even in the high traffic period.
+
+Vertical Scaling:- It involves the increase of recources (like CPU, RAM, storage) of existing server. Which adds more power to single 
+                   machine to  handle more load. If you have limited resources to invest in adition server. It is generally helpful for 
+                   samller websites or for those who donot  expect high traffic.
+
+Horizontal Scaling:- It involves in  adding morew servers to the infrastructure. It's like having multiple machine to share the           
+                     load. It is genrally used when a website experience rapid groth and high traffic and a single server cannot handle the 
+                     load properly.. By adding more server you distribute the load evenly which elentually results in better        
```

Comment:
Refer https://www.cloudzero.com/blog/horizontal-vs-vertical-scaling for more details and better understanding.